### PR TITLE
Rendering fix

### PR DESCRIPTION
**Pull request by kamil-matula**
Scrolling down the list will result in spotting cards with placeholders - these pokemons has not been rendered yet.

image

### CODE REVIEW

File: src/components/PokemonCard.js [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+import { PokemonDataContext } from '../contexts/PokemonDataContext';
 
 export default function PokemonCard({ onPress, id, color, name, sprite, types }) {
   const colors = useTheme().colors;
-  const cardColor = colors.pokemon.background[color];
+
+  const { updatePokemonObject } = useContext(PokemonDataContext);
+  useEffect(() => {
+    pokemonByNameOrNumber(id).then((pokemon) => updatePokemonObject(pokemon));
+  }, []);
```
Comment (Blazevarjo) [inline src/components/PokemonCard.js]: Should be pure component so any effects or functionality shouldn't appear here

File: src/components/PokemonCard.js [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
 
   return (
-    <Surface style={[styles.container, { backgroundColor: cardColor }]}>
+    <Surface style={[styles.container, { backgroundColor: colors.pokemon.background[color] }]}>
       <TouchableRipple borderless={true} style={styles.touch} onPress={() => onPress()}>
```
Comment (Blazevarjo) [inline src/components/PokemonCard.js:22]: It's better to pass prop with color

File: src/screens/main/Home.js [hunk 3]
```diff
@@ -10,11 +10,13 @@ export default function Home() {
...
       <PokemonCard
         onPress={() => {
-          navigation.navigate('Details');
-          updateCurrentPokemonId(id);
+          if (loaded == true) {
+            navigation.navigate('Details');
+            updateCurrentPokemonId(id);
+          }
         }}
         id={id}
```
Comment (Blazevarjo) [inline src/screens/main/Home.js]: If statement in render item also reduce performance in list rendering

Comment (Blazevarjo) [reply src/screens/main/Home.js]: Also 
```
if (loaded==true)
```
should be changed to 
```
if(loaded)
```

File: src/contexts/PokemonDataContext.js [hunk 4]
```diff
@@ -36,7 +36,13 @@ export const PokemonDataProvider = ({ children }) => {
     }
   }, [filters]);
 
-  const resetCurrentPokemon = () => setCurrentPokemon(null);
+  const resetCurrentPokemon = () => setCurrentPokemonId(null);
+
+  const updatePokemonObject = (newPokemonObject) => {
+    let index = pokemons.indexOf(pokemons.find((pokemon) => pokemon.id == newPokemonObject.id));
+    pokemons[index] = newPokemonObject;
+    // TO DO: refresh item
```
Comment (Blazevarjo) [inline src/contexts/PokemonDataContext.js]: State should be treated as immutable.

That's cleaner approach
```
pokemons.findIndex((pokemon) => pokemon.id == newPokemonObject.id)
```

**Review by Blazevarjo — CHANGES_REQUESTED** [review]
I think that this approach  affects poor performance of flatlist. Better way to tweak performance would be to apply Pure Components to pokemon card, cache data and change fetching whole data to smaller batches. We can also think about optimizing images, but first we should focus on developing major functionality, then will be time for optimizing.

In my opinion we should abondon this PR at least for a while

**Review by piotr-wawro — APPROVED** [review]

**Pull request merged**